File: backend/Models/Pagination/PagedResult.cs
namespace Cdr.Api.Models.Pagination;

public class PagedResult<T>
{
    public int PageIndex { get; set; }

    public int PageSize { get; set; }

    public long TotalCount { get; set; }

    public int TotalPages { get; set; }

    public IList<T> Items { get; set; }
}

File: frontend/src/core/models/PagedResult.ts
class PagedResult<T> implements PagedResultModel<T> {
  pageIndex: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  items: T[];

  constructor(data: PagedResultModel<T>) {
    this.pageIndex = data.pageIndex;
    this.pageSize = data.pageSize;
    this.totalCount = data.totalCount;
    this.totalPages = data.totalPages;
    this.items = data.items ?? [];
  }
}
